### DISCUSSION TITLE
Contributing to packages from others in the group

### DISCUSSION THREAD

**Discussion by younesStrittmatter** [AutoResearch/autora#373] [Core Contributions] [answered]
When contributing to our packages (especially), the ones where we are not the main author, should we adapt the workflow for autora?

Create an issue → Create a branch → Push the changes → Create a pull request

**Comment by musslick** [comment]
Are you referring to the "contribution repo" or the sub-packages?

**Reply by younesStrittmatter** [reply]
For example, I changed the urls in the autora-core package that John is working on, instead of directly just commiting and pushing it might be better if we do the full issue->branch->pull request thing. So John can see what I have changed and if he approves.

**Reply by musslick** [reply]
Let's do that. We should report the workflow in the Readme though...

**Reply by hollandjg** [reply, marked as answer]
I'm not so sure we need such a long chain. I think the key bit is just to create a pull request, everything else is sort of implied. We can just link here in our overall contribution guide: https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/about-pull-requests#

**Reply by hollandjg** [reply]
... and make sure that all the repositories are suitably protected to avoid accidental non-PR changes.
